transfers/amclient.py

```diff
@@ -0,0 +1,518 @@
+                'backupCount': 2,
+                'maxBytes': 10 * 1024,
+            },
+        },
+        'loggers': {
+            'amclient': {
+                'level': log_level,
+                'handlers': ['file'],
+            },
+            'requests.packages.urllib3': {
+                'level': log_level,
+                'handlers': ['file'],
+            }
+        },
+    }
+    logging.config.dictConfig(CONFIG)
+
+
+class AMClient:
+
+    def __init__(self,
+                 ss_url,
+                 ss_user_name,
+                 ss_api_key,
+                 am_url,
+                 am_user_name,
+                 am_api_key,
+                 output_mode,
+                 ts_location_uuid,
+                 ts_path_prefix,
+                 aip_uuid,
+                 dip_uuid):
+
+        self.ss_url = ss_url
+        self.ss_user_name = ss_user_name
+        self.ss_api_key = ss_api_key
+        self.am_url = am_url
+        self.am_user_name = am_user_name
+        self.am_api_key = am_api_key
+        self.output_mode = output_mode
+        self.ts_location_uuid = ts_location_uuid
+        self.ts_path_prefix = ts_path_prefix
+        self.aip_uuid = aip_uuid
+        self.dip_uuid = dip_uuid
+
+    def __getattr__(self, name):
+        if name.startswith('print_'):
+            method = name.replace('print_', '', 1)
+            if method == 'transferables':
+                ret = getattr(self, method)(True)
+            else:
+                ret = getattr(self, method)()
+            self.stdout(ret)
+        else:
+            raise AttributeError('AMClient has no method {0}'.format(name))
+
+    def _call_url_json(self, url, params):
+        """Helper to GET a URL where the expected response is 200 with JSON.
+
+        :param str url: URL to call
+        :param dict params: Params to pass to requests.get
+        :returns: Dict of the returned JSON or None
+        """
+        LOGGER.debug('URL: %s; params: %s;', url, params)
+        response = requests.get(url, params=params)
+        LOGGER.debug('Response: %s', response)
+        if not response.ok:
+            LOGGER.warning('Request to %s returned %s %s', url,
+                           response.status_code, response.reason)
+            LOGGER.debug('Response: %s', response.text)
+            return None
+        try:
+            return response.json()
+        except ValueError:  # JSON could not be decoded
+            LOGGER.warning('Could not parse JSON from response: %s',
+                           response.text)
+            return None
+
+    def _am_auth(self):
+        return {
+            'username': self.am_user_name,
+            'api_key': self.am_api_key,
+        }
+
+    def _ss_auth(self):
+        return {
+            'username': self.ss_user_name,
+            'api_key': self.ss_api_key
+        }
+
+    def hide_transfer(self, transfer_uuid):
+        """GET transfer/<UUID>/delete/."""
+        return requests.delete(
```

Comment:
Ok. I've modified call_url_json to a func (no need to be a method) and it takes a method string param and calls requests.request instead of requests.get and it logs the method.